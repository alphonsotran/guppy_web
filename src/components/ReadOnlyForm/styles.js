import styled, { css } from 'styled-components';

export const Input = styled.input`
  border: 0;
  border-bottom: 1px solid black;
  display: block;
  font-family: Brand, sans-serif;
  font-size: 1rem;
  padding: 0.5rem;
  margin-top: 0.5rem;
  width: 18rem;

  ::placeholder {
    color: 'gray';
  }

  ::-ms-clear {
    display: none;
  }
`;

export const ReadOnlyWrapper = styled.div`
  display: flex;
  flex: 1;
  margin-top: 2rem;
  align-self: center;
`;

export const Button = styled.button`
  margin-top: 1rem;
  border: 1px solid black;
  border-radius: 1rem;
  background-color: lightgray;
  margin-top: 1rem;
  padding: 0.8rem;
`;

export const Label = styled.label`
  font-weight: 700;
`;

export const ButtonWrap = styled.div`
  display: flex;
  flex: 1;
  flex-direction: row-reverse;
  justify-content: space-between;
  align-content: center;
`;
