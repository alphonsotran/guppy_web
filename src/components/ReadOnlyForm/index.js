import React from 'react';

import { Input, ReadOnlyWrapper, Button, Label } from './styles';

const ReadOnlyForm = ({ value, status }) => {
  const copyToClipboard = (e) => {
    e.preventDefault();
    navigator.clipboard.writeText(value);
  };

  if (!status) {
    return null;
  }

  return (
    <ReadOnlyWrapper>
      <form onSubmit={copyToClipboard}>
        <Label htmlFor="link">
          Guppy URL
          <Input value={value} readOnly />
        </Label>
        <Button>Copy</Button>
      </form>
    </ReadOnlyWrapper>
  );
};

export default ReadOnlyForm;
