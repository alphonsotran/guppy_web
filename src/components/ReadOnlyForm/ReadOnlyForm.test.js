import React from 'react';
import { shallow } from 'enzyme';
import ReadOnlyForm from './index';

it('expects to render ReadOnlyForm component', () => {
  const mockValue = 'https://localhost:5000/kdfjq2';
  const mockStatus = true;
  expect(
    shallow(<ReadOnlyForm value={mockValue} status={mockStatus} />),
  ).toMatchSnapshot();
});
