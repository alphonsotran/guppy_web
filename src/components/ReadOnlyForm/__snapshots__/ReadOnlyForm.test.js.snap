// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render ReadOnlyForm component 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <styled.label>
      Guppy URL
      <styled.input
        readOnly={true}
        value="https://localhost:5000/kdfjq2"
      />
    </styled.label>
    <styled.div>
      <styled.button>
        Copy
      </styled.button>
    </styled.div>
  </form>
</styled.div>
`;
